From 742335d0504bc03d248944d9eedcc7fb6b4c44aa Mon Sep 17 00:00:00 2001
From: Johan Malm <jgm323@gmail.com>
Date: Mon, 2 Jan 2023 18:29:03 +0000
Subject: [PATCH] widgets: fix button-release bug

Do not execute widget action on button release if pointer has moved
outside widget area since pressed

Reported-by: @RabidPistachio

Fixes issue #189
---
 src/jgmenu.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/jgmenu.c b/src/jgmenu.c
index c1243f3..727c6ce 100644
--- a/src/jgmenu.c
+++ b/src/jgmenu.c
@@ -1869,11 +1869,12 @@ static void mouse_release(XEvent *e)
 
 	/* left-click */
 	if (ev->button == Button1) {
-		char *ret;
+		char *ret = NULL;
 
 		/* widgets */
 		widgets_set_pointer_position(mouse_coords.x, mouse_coords.y);
-		ret = widgets_get_selection_action();
+		if (widgets_mouseover() == 1)
+			ret = widgets_get_selection_action();
 		if (ret && ret[0] != '\0') {
 			action_cmd(ret, NULL);
 			return;
